drivers/soundwire/bus.c

```diff
@@ -950,9 +949,8 @@ int sdw_bus_prep_clk_stop(struct sdw_bus *bus)
 			ret = sdw_slave_clk_stop_prepare(slave,
 							 SDW_CLK_STOP_MODE0,
 							 true);
-			if (ret < 0) {
-				dev_err(&slave->dev,
-					"pre-prepare failed:%d", ret);
+			if (ret < 0 && ret != -ENODATA) {
```

Comment:
i'll have to review this, thanks for the comment.

Reply:
@lyakh you are correct that in the loops we ignore the -ENODATA and keep going, so we should not do a return ret later. I fixed this by using return 0 (which I already used for the clock-exit btw, so that makes the code even more aligned than before).

for the sdw_cdns_clock_stop(), I see all cases checking for -ENODATA. The only remaining test is 

````c
	ret = cdns_set_wait(cdns, CDNS_MCP_STAT,
			    CDNS_MCP_STAT_CLK_STOP,
			    CDNS_MCP_STAT_CLK_STOP);
	if (ret < 0)
		dev_err(cdns->dev, "Clock stop failed %d\n", ret);
````

but in that case it's a register write on the host side, i.e. there is no command sent on the bus so -ENODATA cannot happen.